pg_ivm.c

```diff
@@ -320,3 +322,30 @@ PgIvmImmvPrimaryKeyIndexId(void)
 	return pg_ivm_immv_pkey_id;
 }
 
+/*
+ * Return the SELECT part of a IMMV
+ */
+Datum
+get_immv_def(PG_FUNCTION_ARGS)
+{
+	Oid	matviewOid = PG_GETARG_OID(0);
+	Relation matviewRel = NULL;
+	Query *query = NULL;
+	char *querystring = NULL;
+
+	matviewRel = table_open(matviewOid, AccessShareLock);
+	/* Make sure IMMV is a table. */
+	Assert(matviewRel->rd_rel->relkind == RELKIND_RELATION);
+
+	query = get_immv_query(matviewRel);
+	if (query == NULL)
+	{
+		table_close(matviewRel, NoLock);
+		PG_RETURN_TEXT_P(cstring_to_text(""));
```

Comment:
Should we return NULL instead of an empty string, as pg_get_viewdef?

Reply:
ok, I fixed it